sdk/Pulumi.Automation/Commands/LocalPulumiCommand.cs

```diff
@@ -0,0 +1,417 @@
+// Copyright 2016-2021, Pulumi Corporation
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Semver;
+using CliWrap;
+using CliWrap.Buffered;
+using Pulumi.Automation.Commands.Exceptions;
+using Pulumi.Automation.Events;
+
+
+namespace Pulumi.Automation.Commands
+{
+    /// <summary>
+    /// Options to configure a <see cref="LocalPulumiCommand"/> instance.
+    /// </summary>
+    public class LocalPulumiCommandOptions
+    {
+        /// <summary>
+        /// The version of the Pulumi CLI to install or the minimum version requirement for an existing installation.
+        /// </summary>
+        public SemVersion? Version { get; set; }
+        /// <summary>
+        /// The directory where to install the Pulumi CLI to or where to find an existing installation.
+        /// </summary>
+        public string? Root { get; set; }
+        /// <summary>
+        /// If true, skips the version validation that checks if an existing Pulumi CLI installation is compatible with the SDK.
+        /// </summary>
+        public bool SkipVersionCheck { get; set; }
+    }
+
+    /// <summary>
+    /// A <see cref="IPulumiCommand"/> implementation that uses a locally installed Pulumi CLI.
+    /// </summary>
+    public class LocalPulumiCommand : IPulumiCommand
+    {
+        // TODO: move to shared place with LocalWorkspace
+        private static string SkipVersionCheckVar = "PULUMI_AUTOMATION_API_SKIP_VERSION_CHECK";
+        private static readonly SemVersion _minimumVersion = new SemVersion(3, 1, 0);
+        private readonly string _command;
+        private SemVersion? _version;
+
+        /// <summary>
+        /// The version of the Pulumi CLI that is being used.
+        /// </summary>
+        public SemVersion? Version { get => _version; }
```

Comment:
Consider getting rid of the `_version` field and just having this:

```suggestion
        public SemVersion? Version { get; }
```